### PR TITLE
update contracts in tutorial to work with Solidity 0.8.0

### PR DESCRIPTION
**Pull request by jmcook1186**
## Description

Noticed this issue has been hanging for a while so had a crack at it - hope that's ok.

Issue #5354 describes a declaration error thrown when the contract in tutorial in the developer docs [here](https://github.com/ethereum/ethereum-org-website/blob/864d77178dee04e80d5f983b3911cc351118fb26/src/content/developers/tutorials/transfers-and-approval-of-erc-20-tokens-from-a-solidity-smart-contract/index.md) is compiled using Solidity 0.8.0.

This is not really a huge problem since the tutorial is written with a clear `pragma Solidity ^0.6.0` statement. As long as the pragma statement remains unchanged at `^0.6.0` the contracts compile perfectly and behave as expected. Therefore, the mods might decide this PR is unnecessary and close it without merging. However, some users might want to use the most up to date Solidity compiler, especially because there have been significant breaking changes between v0.6 and v0.8.

This PR includes an amended tutorial page with contracts that compile and run as expected using Solidity v0.8. Specifically the following actions were taken:

1) Removed redundant access modifiers from the constructors
2) The v0.8 compiler no longer implicitly converts `address` to `address payable`. Contract amended accordingly.
3) The events that were declared twice in the previous version are now declared once in the interface definition and inherited by the contract, fixing the `DeclarationError`.
4) Solidity 0.8 includes built-in overflow checking that makes SafeMath redundant. Removed in updated version.
5) Added an explicit description of the token approval process to the tutorial as I thought this was opaque in the original. Especially because it requires instantiation of the token contract at the address where it was deployed during the DEX contract creation. 
6) Added a note to the prerequisite page explaining that SafeMath is unneccessary for Solidity 0.8.

Original and updated tutorial contracts tested using Brownie v1.14.6. Site updates checked by hosting locally.

This PR is kind of a patch stimulated by issue #5354 but perhaps a completely fresh tutorial on this topic might be good soon?

## Related Issue

#5354

### CODE REVIEW

File: src/content/developers/tutorials/understand-the-erc-20-token-smart-contract/index.md [hunk 1]
```diff
@@ -181,6 +181,8 @@ contract ERC20Basic is IERC20 {
     }
 }
 
+// note that the SafeMath library is not required for Solidity v0.8 and above.
+
```
Comment (minimalsm) [inline src/content/developers/tutorials/understand-the-erc-20-token-smart-contract/index.md:185]: Should we just remove SafeMath entirely here?

Maybe point to it as further reading?

File: src/content/developers/tutorials/transfers-and-approval-of-erc-20-tokens-from-a-solidity-smart-contract/index.md [hunk 2]
```diff
@@ -298,14 +289,14 @@ contract DEX {
...
-    constructor() public {
+    constructor() {
         token = new ERC20Basic();
     }
 
     function buy() payable public {
         uint256 amountTobuy = msg.value;
         uint256 dexBalance = token.balanceOf(address(this));
-        require(amountTobuy > 0, "You need to send some ether");
+        require(amountTobuy > 0, "You need to send some Ether");
```
Comment (minimalsm) [inline src/content/developers/tutorials/transfers-and-approval-of-erc-20-tokens-from-a-solidity-smart-contract/index.md:299]: As per [our style-guide:](https://ethereum.org/en/contributing/style-guide/#ether)

> _Ether is a common noun and should not be capitalized unless at the beginning of a sentence. ETH, on the other hand, is a currency abbreviation (and ticker symbol) and should always be capitalized._

```suggestion
        require(amountTobuy > 0, "You need to send some ether");
```

File: src/content/developers/tutorials/transfers-and-approval-of-erc-20-tokens-from-a-solidity-smart-contract/index.md [hunk 3]
```diff
@@ -159,7 +143,7 @@ To keep things simple, we just exchange 1 token for 1 Wei.
 function buy() payable public {
     uint256 amountTobuy = msg.value;
     uint256 dexBalance = token.balanceOf(address(this));
-    require(amountTobuy > 0, "You need to send some ether");
+    require(amountTobuy > 0, "You need to send some Ether");
```
Comment (minimalsm) [inline src/content/developers/tutorials/transfers-and-approval-of-erc-20-tokens-from-a-solidity-smart-contract/index.md:146]:
```suggestion
    require(amountTobuy > 0, "You need to send some ether");
```

File: src/content/developers/tutorials/transfers-and-approval-of-erc-20-tokens-from-a-solidity-smart-contract/index.md [hunk 4]
```diff
@@ -1,7 +1,7 @@
 ---
 title: Transfers and approval of ERC-20 tokens from a solidity smart contract
 description: How to use a smart contract to interact with a token using the Solidity language
-author: "jdourlens"
+author: "jdourlens, updated 2022 by jcook"
```
Comment (minimalsm) [inline src/content/developers/tutorials/transfers-and-approval-of-erc-20-tokens-from-a-solidity-smart-contract/index.md:4]: Unfortunately, our current design system doesn't support this. We should definitely explore this in our tutorial section revamp.

```suggestion
author: "jdourlens"
```

File: src/content/developers/tutorials/understand-the-erc-20-token-smart-contract/index.md (hunk 1 above)
Comment (jmcook1186) [reply src/content/developers/tutorials/understand-the-erc-20-token-smart-contract/index.md:185]: @minimalsm sure - we can. I was trying to be minimal - the contracts on that page are also using Solidity v0.6.0 so they need to be updated for Solidity v0.8.0 too if we want to remove SafeMath altogether. Happy to do this if you think it is neater to have both tutorials using v0.8.0.

**Review by minimalsm — APPROVED** [review]

**Comment by minimalsm** [comment]
Thanks again @jmcook1186 🎉

**Pull request merged**